src/react-components/src/components/02-molecules/SearchBar/SearchBar.tsx

```diff
@@ -28,14 +33,28 @@ export interface SearchBarProps {
 
 export default function SearchBar(props: SearchBarProps) {
 
-  const { searchBarId, buttonId, dropdownId, dropdownOptions, selectedField, searchValue, placeholderText, textFieldAriaLabel,
-    hasError, errorMessage, selectBlurHandler, searchSubmitHandler, selectChangeHandler, searchChangeHandler } = props;
+  const { blockName,
+    searchBarId, buttonId,
+    searchBarAriaLabel, searchBarAriaLabelledBy,
+    dropdownId, dropdownOptions, dropdownAriaLabel,
+    selectedField, searchValue,
+    placeholderText,
+    helperVariant,
+    hasError, errorMessage,
```

Comment:
The following is more of an idea/example rather than something you _have_ to do.

In SimplyE we have a `FetchErrorData` interface defined as:
```
export interface FetchErrorData {
  status: number;
  response: string;
  url: string;
}
```

This is for making API requests and the object conforms to what the server returns. In components we check (as an example):
```
// ...
error: FetchErrorData
// ...

{ error && <ErrorMessage message={error.response} /> }
```

I'm not sure how RN is structured, but it might be easier to define an interface for errors you expect so you don't have to (1) pass in two props to check and render an error message (2) throw an error if one variable is passed but not the other. This interface pattern can also be used for the other components that use this two-prop pattern. But, again, something to think about.

Reply:
I'm gonna keep this as is and keep this in mind - Right now errors in RN are pretty dumb but if the error component becomes a problem while trying to implement, I'll come back to it.  

The thing I'm quickly learning is that passing a raw `url` is almost always going to cause problems in RN ... but also this is going to be a recurring problem so we do want to come up with a solid solution.